crates/pallets/starknet/src/utils.rs

```diff
@@ -57,7 +70,8 @@ pub fn execute_txs_and_rollback<T: pallet::Config>(
                 PlaceHolderErrorTypeForFailedStarknetExecution
             });
 
-            execution_results.push((result, cached_state.to_state_diff()));
+            let state_diff = if with_state_diff { Some(cached_state.to_state_diff()) } else { None };
+            execution_results.push((result, state_diff));
```

Comment:
The problem is that even with `with_state_diff == false` it still uses `CachedBlockifierStateAdapter` which means extra work (maintaining the cache) that won't be used.

You have to either have two methods. Or internally branch to using `CachedBlockifierStateAdapter`/`BlockifierStateAdapter` depending on the bool

Reply:
Fixed, ended up going with another function as we want to keep static sized traits for execution-related functions imo